packages/filecoin-actor-utils/src/utils/params.ts

```diff
@@ -32,3 +34,30 @@ export const describeMessageParams = (
   // Return the described params
   return dataType
 }
+
+// TODO: (nice to have) enable typechain artifact as a typescript generic here so the described params are typed?
+export const describeFEVMMessageParams = (
+  messageParams: string,
+  abi: ABI
+): DataType => {
+  const iface = new ethers.utils.Interface(abi)
+  const paramsHex = cborToHex(messageParams)
+  const parsed = iface.parseTransaction({ data: paramsHex })
+
+  // TODO: how to handle solidity structs as inputs? I believe structs are encoded as `tuple` type https://docs.soliditylang.org/en/v0.5.3/abi-spec.html#mapping-solidity-to-abi-types
+  // TODO: unstandardized capitalization from built-in actors
+  const children = parsed.functionFragment.inputs.reduce<
+    Record<string, DataType>
+  >((accum, ele) => {
+    return {
+      ...accum,
+      [ele.name]: {
+        Name: ele.name,
+        Type: ele.type as Type,
```

Comment:
@navFooh - I wasn't sure the best way to handle `Type` here. Generally I see two ways:

1. We add all the solidity types to our `Types` and handle them throughout the react-components in our `DataType` components
2. We transform solidity abi types into our predefined `Types`, so for instance, `uint256` and `uint32` would just be `number`

What do you think?